fud2/fud-core/src/run.rs

```diff
@@ -87,6 +87,77 @@ impl EmitBuild for EmitBuildFn {
     }
 }
 
+/// A config variable.
+#[derive(Debug, Clone)]
+pub enum ConfigVar {
+    /// The key for the config variable.
+    Var(String),
+    /// The key for the config variable followed by the value it should be if the key is not found.
+    VarOr(String, String),
+}
+
+/// The data required to emit a single op.
+pub struct OpEmitData {
+    /// The name of the op.
+    pub op_name: String,
+
+    /// The commands run whenever the op run. Each of these must be run in order in a context where
```

Comment:
Again, let's avoid the abstraction here and just say that these are the shell commands that will be emitted as the body of the Ninja rule.